src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java

```diff
@@ -0,0 +1,87 @@
+package com.internet.shop.web.filters;
+
+import com.internet.shop.library.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.interfaces.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorisationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/order/complete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/product/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/manage", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("shopping-cart/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/admin/details", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders", List.of(Role.RoleName.ADMIN));
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String requestedUrl = req.getServletPath();
+
+        if (protectedUrls.get(requestedUrl) == null) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null || userService.get(userId) == null) {
+            resp.sendRedirect("/login");
+            return;
```

Comment:
we did this check in Authentification Filter
```suggestion

```

Reply:
forgot to make changes in Authentication filter, thanks